File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.java
class Solution {
    class FrequencyPair implements Comparable<FrequencyPair>{
        private int freq;
        private int value;
        public FrequencyPair(int f, int v){
            freq = f;
            value = v;
        }
        public int compareTo(FrequencyPair other){
            return Integer.compare(freq, other.freq);
        }
    }

    public int[] topKFrequent(int[] nums, int k) {
        HashMap<Integer, Integer> freq = new HashMap<>();
        for(int i : nums){
            freq.merge(i, 1, Integer::sum);
        }
        PriorityQueue<FrequencyPair> pq = new PriorityQueue<>(Collections.reverseOrder());
        for(int i : freq.keySet()){
            pq.add(new FrequencyPair(freq.get(i), i));
        }
        int[] ans = new int[k];
        for (int i=0; i<k; i++){
            ans[i]=pq.poll().value;
        }
        return ans;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        mult = 1
        zeroCount = 0
        for x in nums:
            if (x!=0):
                mult *= x
            else:
                zeroCount+=1
        return [0 if zeroCount>1 or (zeroCount==1 and x!=0) else int(mult/(1 if x==0 else x)) for x in nums]

        